### PR TITLE
Improve I18n use in payment method views

### PR DESCRIPTION
**Pull request by Murph33**
Use model name and attribute translation.

This is part of an ongoing effort to improve I18n use in solidus as discussed in #735.

### CODE REVIEW

File: backend/app/views/spree/admin/payment_methods/_form.html.erb [hunk 1]
```diff
@@ -25,15 +25,15 @@
         <div class="info warning js-gateway-settings-warning"><%= Spree.t(:provider_settings_warning) %></div>
       </div>
       <div data-hook="display" class="field">
-        <%= label_tag :payment_method_display_on, Spree.t(:display) %>
+        <%= label_tag :payment_method_display_on, Spree::PaymentMethod.human_attribute_name(:display_on) %>
```
Comment (mamhoff) [inline backend/app/views/spree/admin/payment_methods/_form.html.erb]: Is there a particular reason you're not using `f.label :display_on` here?

File: backend/app/views/spree/admin/payment_methods/_form.html.erb [hunk 2]
```diff
@@ -25,15 +25,15 @@
...
         <%= select(:payment_method, :display_on, Spree::PaymentMethod::DISPLAY.collect { |display| [Spree.t(display), display == :both ? nil : display.to_s] }, {}, {:class => 'select2 fullwidth'}) %>
       </div>
       <div data-hook="auto_capture" class="field">
-        <%= label_tag :payment_method_auto_capture, Spree.t(:auto_capture) %>
+        <%= label_tag :payment_method_auto_capture, Spree::PaymentMethod.human_attribute_name(:auto_capture) %>
```
Comment (mamhoff) [inline backend/app/views/spree/admin/payment_methods/_form.html.erb]: `f.label :auto_capture`?

File: backend/app/views/spree/admin/payment_methods/index.html.erb [hunk 3]
```diff
@@ -24,10 +24,10 @@
     </colgroup>
     <thead>
       <tr data-hook="admin_payment_methods_index_headers">
-        <th><%= Spree.t(:name) %></th>
-        <th><%= Spree.t(:provider) %></th>
-        <th><%= Spree.t(:display) %></th>
-        <th><%= Spree.t(:active) %></th>
+        <th><%= Spree::PaymentMethod.human_attribute_name(:name) %></th>
+        <th><%= Spree::PaymentMethod::human_attribute_name(:type) %></th>
```
Comment (mamhoff) [inline backend/app/views/spree/admin/payment_methods/index.html.erb]: This has `::` before the method call, where you want  a single `.` (also the two lines below).

**Comment by mamhoff** [comment]
The tests failed, I suppose because of `phantomjs` crashing. You can re-trigger the build by amending the commit and force-pushing. ;)

File: core/config/locales/en.yml [hunk 4]
```diff
@@ -720,7 +726,7 @@ en:
     average_order_value: Average Order Value
     avs_response: AVS Response
     back: Back
-    back_end: Backend
+    back_end: Back End
```
Comment (mamhoff) [inline core/config/locales/en.yml]: This translation is not used in this PR, and I'm not sure whether `Back End` is actually more correct than `Backend`.

Comment (Murph33) [inline core/config/locales/en.yml]: I think I saw that translation used on a page so I tossed in the change but after [investigating](https://dzone.com/articles/blogging-programmers-style) it seems like the old choice of `Front End` and `Backend` isn't as strange as I thought.

**Comment by mamhoff** [comment]
:+1:

**Comment by jhawthorn** [comment]
:+1:

**Pull request merged**